src/libAtomVM/nifs.c

```diff
@@ -1396,24 +1399,73 @@ term nif_erlang_universaltime_0(Context *ctx, int argc, term argv[])
     term time_tuple = term_alloc_tuple(3, &ctx->heap);
     term date_time_tuple = term_alloc_tuple(2, &ctx->heap);
 
+    term_put_tuple_element(date_tuple, 0, term_from_int32(1900 + broken_down_time->tm_year));
+    term_put_tuple_element(date_tuple, 1, term_from_int32(broken_down_time->tm_mon + 1));
+    term_put_tuple_element(date_tuple, 2, term_from_int32(broken_down_time->tm_mday));
+
+    term_put_tuple_element(time_tuple, 0, term_from_int32(broken_down_time->tm_hour));
+    term_put_tuple_element(time_tuple, 1, term_from_int32(broken_down_time->tm_min));
+    term_put_tuple_element(time_tuple, 2, term_from_int32(broken_down_time->tm_sec));
+
+    term_put_tuple_element(date_time_tuple, 0, date_tuple);
+    term_put_tuple_element(date_time_tuple, 1, time_tuple);
+
+    return date_time_tuple;
+}
+
+term nif_erlang_universaltime_0(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    UNUSED(argv);
+
     struct timespec ts;
     sys_time(&ts);
 
     struct tm broken_down_time;
-    gmtime_r(&ts.tv_sec, &broken_down_time);
+    return build_datetime_from_tm(ctx, gmtime_r(&ts.tv_sec, &broken_down_time));
+}
 
-    term_put_tuple_element(date_tuple, 0, term_from_int32(1900 + broken_down_time.tm_year));
-    term_put_tuple_element(date_tuple, 1, term_from_int32(broken_down_time.tm_mon + 1));
-    term_put_tuple_element(date_tuple, 2, term_from_int32(broken_down_time.tm_mday));
+term nif_erlang_localtime(Context *ctx, int argc, term argv[])
+{
+    char *tz;
+    if (argc == 1) {
+        int ok;
+        tz = interop_term_to_string(argv[0], &ok);
```

Comment:
Is `tz` getting free’d in this function?

Reply:
It wasn't. Thank you.